pkg/api/aim/runs.go

```diff
@@ -915,6 +916,111 @@ func DeleteRun(c *fiber.Ctx) error {
 	})
 }
 
+// UpdateRun will update the run name, description, and lifecycle stage
+func UpdateRun(c *fiber.Ctx) error {
+	params := struct {
+		ID string `params:"id"`
+	}{}
+	if err := c.ParamsParser(&params); err != nil {
+		return fiber.NewError(fiber.StatusUnprocessableEntity, err.Error())
+	}
+
+	var update request.UpdateRun
+	if err := c.BodyParser(&update); err != nil {
+		return fiber.NewError(fiber.StatusUnprocessableEntity, err.Error())
+	}
+
+	// TODO this code should move to service
+	run := models.Run{ID: params.ID}
+	runRepo := repositories.NewRunRepository(database.DB.DB)
+	var err error
+	if update.Archived != nil {
+		if *update.Archived {
+			err = runRepo.Archive(c.Context(), &run)
+		} else {
+			err = runRepo.Restore(c.Context(), &run)
+		}
+	}
+	if err != nil {
+		return fiber.NewError(fiber.StatusInternalServerError,
+			fmt.Sprintf("unable to archive/restore run %q: %s", params.ID, err))
+	}
+
+	if update.Name != nil {
+		run.Name = *update.Name
+		err = database.DB.DB.Transaction(func(tx *gorm.DB) error {
+			if err := runRepo.UpdateWithTransaction(c.Context(), tx, &run); err != nil {
+				return err
+			}
+			return nil
+		})
+	}
+	if err != nil {
+		return fiber.NewError(fiber.StatusInternalServerError,
+			fmt.Sprintf("unable to update run %q: %s", params.ID, err))
+	}
+
+	return c.JSON(fiber.Map{
+		"id":     params.ID,
+		"status": "OK",
+	})
+}
+
+func ArchiveBatch(c *fiber.Ctx) error {
+	var ids []string
+	if err := c.BodyParser(&ids); err != nil {
+		return fiber.NewError(fiber.StatusUnprocessableEntity, err.Error())
+	}
+
+	// TODO this code should move to service
+	runRepo := repositories.NewRunRepository(database.DB.DB)
+	if err := database.DB.DB.Transaction(func(tx *gorm.DB) error {
+		for _, id := range ids {
+			run := models.Run{ID: id}
+			var err error
+			if c.Query("archive") == "true" {
+				err = runRepo.Archive(c.Context(), &run)
+			} else {
+				err = runRepo.Restore(c.Context(), &run)
+			}
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
```

Comment:
I think that `RunRepository` should be the one processing a batch of archive/restore requests. This would avoid making n DB calls for n runs. Also I don't think that this transaction actually works, as `tx` is not passed to the `Archive`/`Restore` functions.

Reply:
ah yes, I see -- so I will add a batch interface for these and then we won't need transactions as it will be a singular

Reply:
Updated with batch sql statements in respository